File: frontend/src/main/frontend/src/app/tech-resource-details-dto.ts
import { TagDTO } from './tag-dto'; 
export class TechResourceDetailsDTO {
    id: number;
    title: string;
    link: string;
    tagDTOs: Array<TagDTO>;
    constructor(id: number, title: string, link: string, tagDTOs: Array<TagDTO>) {
        this.id = id;
        this.title = title;
        this.link = link;
        this.tagDTOs = tagDTOs;
    }
}

File: backend/src/main/java/org/bbelovic/techresourcetracker/TechResourceDetailsDTO.java
package org.bbelovic.techresourcetracker;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Collection;

public record TechResourceDetailsDTO(@JsonProperty("id") long id, @JsonProperty("username") String username,
                                     @JsonProperty("title") String title, @JsonProperty("link") String link,
                                     @JsonProperty("tagDTOs") Collection<TagDTO> tagDTOs) {
}
